Update optional dependency to coap-message 0.2

chrysn opened the pull request:
The coap-message traits had to undergo a breaking transition in order to allow the implementation of composable message handlers.

As this affects a non-default feature (that, moreover, probably has not seen any uptake outside in my applications), it probably doesn't warrant a semver update.

---

By the way, now that https://gitlab.com/chrysn/coap-message-demos is out there, it may be worth considering a cutback on the example's features (with a pointer to "see X for all you can do with this"), especially since that includes an example based on coap-lite. If you want, I can add the changes here because the changes are related (in the sense that the update helps in doing the demos properly).

chrysn committed:
Update coap-message traits

The traits crate underwent an incompatible update that allows better
composability.

chrysn committed:
example: Use updated coap-handler idiom

martindisch commented:
Hi, sorry for the late reply.

Absolutely, we can push this out as a non-breaking change.
And yes, there's no need to maintain the full example in two places, go ahead and add your changes if that makes life easier for you.

I do have one question that I'm sure you've already considered: what's the reason for allowing the alpha versions for `coap-handler` and `coap-message`?

chrysn committed:
Merge branch 'master' into coap-message-0.2

chrysn committed:
examples: Remove complexity

There is a dedicated set of examples out now; no need to pull in
complexity to this example, which mainly shows that coap_message & co
can be used together with this crate.

chrysn commented:
Hm, and then I missed yours. Updated now to latest master (just merged master in, no further changes required, things still work).

On the alpha versions, it's mainly because I published them and they were good and do work for these purposes. (It's a complex set of interdependent crates, and for big changes I prefer to do a full test run where everything is released piece by piece as alpha, then I can take all the crate patches out and let things build against the published versions. Mostly, I then just released the same code again as full version, just with any typos or similar non-breaking things fixed.)

I've stripped down the example as per your message.

Please review, and merge if suitable.

martindisch commented:
Great, will push this out as 0.5.1 right away. Thank you!

chrysn commented:
Thanks a bunch, this made coap-message-demos uploadable.
